src/hooks/useBlocksInfo.ts

```diff
@@ -26,41 +27,36 @@ interface Props {
 }
 
 const useBlocksInfo = ({ isApiReady, api, chain }: Props) => {
-  const [bestBlock, setBestBlock] = useState('');
-  const [bestBlockFinalized, setBestBlockFinalized] = useState('');
+  const [bestBlock, setBestBlock] = useMountedState('');
+  const [bestBlockFinalized, setBestBlockFinalized] = useMountedState('');
 
   useEffect(() => {
-    if (!api || !isApiReady || !chain) {
-      setBestBlock('');
-      setBestBlockFinalized('');
-      return;
-    }
-
-    let unsubscribeBestNumber: () => void;
-    api.derive.chain
-      .bestNumber((res) => {
-        setBestBlock(res.toString());
-      })
-      .then((unsub) => {
-        unsubscribeBestNumber = unsub;
-      })
-      .catch((e) => logger.error('error reading blocks', e));
-
-    let unsubscribeBestNumberFinalized: () => void;
-    api.derive.chain
-      .bestNumberFinalized((res) => {
-        setBestBlockFinalized(res.toString());
-      })
-      .then((unsub) => {
-        unsubscribeBestNumberFinalized = unsub;
-      })
-      .catch((e) => logger.error('error reading blocks', e));
+    let unsubscribeBestNumber: Promise<VoidFn>;
+    let unsubscribeBestNumberFinalized: Promise<VoidFn>;
+    if (isApiReady && chain) {
```

Comment:
I find it better to early-exit (like in previous code) instead of keep indenting/nesting the code with every condition.